cap-primitives/src/yanix/common/fs/open_unchecked.rs

```diff
@@ -27,8 +24,15 @@ pub(crate) fn open_unchecked(
         path.as_ref()
     };
 
-    unsafe {
-        let fd = openat(start.as_raw_fd(), path, oflags, mode)?;
-        Ok(fs::File::from_raw_fd(fd))
+    let err = unsafe {
+        match openat(start.as_raw_fd(), path, oflags, mode) {
+            Ok(fd) => return Ok(fs::File::from_raw_fd(fd)),
+            Err(err) => err,
+        }
+    };
+    match err.raw_os_error() {
+        Some(libc::ELOOP) | Some(libc::EMLINK) => Err(OpenUncheckedError::SymlinkDisallowed),
+        None => Err(OpenUncheckedError::Io(err)),
+        Some(e) => Err(OpenUncheckedError::Io(io::Error::from_raw_os_error(e))),
```

Comment:
These two lines can just be `_ => OpenUncheckedError::Io(err),`, so that we pass through any extra error context that might be present in `err`.